### ISSUE TITLE
Move sinatra to a full dependency

### ISSUE THREAD

**Issue by mperham** [mperham/sidekiq#3042]
Sinatra has been an optional dependency for Sidekiq from the start.  Recent changes around Rack 2.0 and Rails 5.0 have made Sinatra versioning unstable, we need to lock down the allowed versions to avoid cryptic error messages.

```diff
+  gem.add_dependency                  'sinatra', '~> 1.4', '>= 1.4.6'
-  gem.add_development_dependency      'sinatra', '~> 1.4', '>= 1.4.6'
```

This will pull in rack, rack-protection and tilt as dependencies but they all have no transitive dependencies.

**Comment by ericboehs** [comment]
This seems to break compatibility with Rails 5.

**Comment by mperham** [comment]
There's no version of Sinatra that supports Rails 5.

**Comment by kgrz** [comment]
@mperham Is this solved now? After the rack 2.0 bump?

**Comment by mperham** [comment]
#3048
